server/src/main/java/org/elasticsearch/lucene/spatial/CartesianShapeCoordinateEncoder.java

```diff
@@ -10,7 +10,7 @@
 
 import org.apache.lucene.geo.XYEncodingUtils;
 
-final class CartesianShapeCoordinateEncoder implements CoordinateEncoder {
+public final class CartesianShapeCoordinateEncoder implements CoordinateEncoder {
```

Comment:
I would really like to keep them package private, why do you need them?

Reply:
These are used in ST_INTERSECTS when creating the geometry doc-values, as well as when creating the Component2DVisitor.

Reply:
But they can refer then a `CoordinateEncoder`? whe don't need to made them public?

Reply:
You should use CoordinateEncoder.CARTESIAN

Reply:
I open https://github.com/elastic/elasticsearch/pull/105502 which hopefully avoid any future issue.

Reply:
Understood. I should have used the available constants instead of creating additional classes. I've approved that PR and will switch to that as soon as it's merged.

Reply:
Pushed a fix, and use the constants instead now (and removed the `public`)